source/compiler-core/slang-json-parser.h

```diff
@@ -0,0 +1,188 @@
+// slang-json-parser.h
+#ifndef SLANG_JSON_PARSER_H
+#define SLANG_JSON_PARSER_H
+
+#include "slang-json-lexer.h"
+
+
+namespace Slang {
+
+class JSONListener
+{
+public:
+        /// Start an object
+    virtual void startObject() = 0;
```

Comment:
An interesting design question here is how to pass source location information through to the listener, so that we can build a JSON AST with location information and properly diagnose errors when mapping from JSON to/from user-defined types.

The simplicity of the current operations is nice, so it might be best if source-location information (and comments, if we want to go that route) could be passed through a side-band interface.